examples/inject/src/main/java/io/helidon/examples/inject/CovarianceExample.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.examples.inject;
+
+import java.util.List;
+
+import io.helidon.service.registry.Service;
+import io.helidon.service.registry.ServiceRegistryManager;
+
+/**
+ * An example that demonstrates covariant lookups.
+ */
+class CovarianceExample {
+
+    private CovarianceExample() {
+    }
+
+    sealed interface Pet permits Cat, Dog {
+        default String name() {
+            return getClass().getSimpleName();
+        }
+    }
+
+    sealed interface Cat extends Pet permits Siamese, Bengal {
+    }
+
+    sealed interface Dog extends Pet permits Boxer, Husky {
+    }
+
+    @Service.Singleton
+    record Siamese() implements Cat {
+    }
+
+    @Service.Singleton
+    record Bengal() implements Cat {
+    }
+
+    @Service.Singleton
+    record Boxer() implements Dog {
+    }
+
+    @Service.Singleton
+    record Husky() implements Dog {
+    }
+
+    @Service.Singleton
+    record Shelter(List<Pet> all, List<Cat> cats, List<Dog> dogs) {
+    }
+
+    public static void main(String[] args) {
+        var registry = ServiceRegistryManager.create().registry();
```

Comment:
The examples should either use the `Services` class, or create a manager and shut it down when finished.
When using method `create()`, the registry manager is not registered with a shutdown hook, so `@PreDestroy` methods would not get called.

Reply:
Should it implement AutoCloseable ?

Reply:
I have replaced most usages with Services.
There are two classes that still use ServiceRegistryManager, I've used a try/finally block in those.

Reply:
I would prefer this fixed, as this will get re-arranged when I run code format, and that is the reason why we have code format specified in the project...